tests/ada_c.cpp

```diff
@@ -52,40 +69,43 @@ TEST(ada_c, getters) {
 }
 
 TEST(ada_c, setters) {
-  ada_url url = ada_parse(
+  std::string input =
       "https://username:password@www.google.com:8080/"
-      "pathname?query=true#hash-exists");
+      "pathname?query=true#hash-exists";
+  ada_url url = ada_parse(input.data(), input.length());
 
   ASSERT_TRUE(ada_is_valid(url));
 
-  ada_set_href(url, "https://www.yagiz.co");
+  ada_set_href(url, "https://www.yagiz.co",
+               std::string("https://www.yagiz.co").length());
   ASSERT_EQ(convert_string(ada_get_href(url)), "https://www.yagiz.co/");
 
-  ada_set_username(url, "new-username");
+  ada_set_username(url, "new-username", std::string("new-username").length());
   ASSERT_EQ(convert_string(ada_get_username(url)), "new-username");
 
-  ada_set_password(url, "new-password");
+  ada_set_password(url, "new-password", std::string("new-password").length());
```

Comment:
Instead of "std::string(...).length()", do "strlen(...)" or the C++ way might be "std::char_traits<char>::length(...)" https://en.cppreference.com/w/cpp/string/char_traits/length

Reply:
You're right 👍